platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -537,12 +538,15 @@ protected void appendContents(TreeFormatter formatter) {
     /**
      * {@inheritDoc}
      *
-     * @implNote Usage: This method should only be called on resolvable configurations and should throw an exception if
+     * @implNote Usage: This method should only be called on resolvable configurations and throws an exception if
      * called on a configuration that does not permit this usage.
      */
     @Override
     public boolean contains(File file) {
-        warnOnInvalidInternalAPIUsage("contains(File)", ProperMethodUsage.RESOLVABLE);
+        if (!isProperUsage(ProperMethodUsage.RESOLVABLE)) {
```

Comment:
💭 Maybe this should use `assertIsResolvable()` as well? I think the others merit custom messages but I wasn't sure about this one.